llvm/lib/Target/AIE/AIEPostPipeliner.cpp

```diff
@@ -190,29 +190,35 @@ void PostPipeliner::computeLoopCarriedParameters() {
   }
 
   // Propagate Earliest upstream, initialize Latest
+  // Unrestricted: last cycle of last stage
+  const int Latest = NCopies * II - 1;
   for (int K = 0; K < NInstr; K++) {
     const int K2 = K + NInstr;
     const int Earliest = Info[K2].Earliest - II;
```

Comment:
Checking: this could be negative

Reply:
Yes, II think this can be negative. In particular, [].Earliest only gets >0 if there's a loop carried dependence. However, this is only used to max, and is int, so it doesn't matter.